tools/interactor-system-func-calls/sys-func-calls/interact/src/basic_interact.rs

```diff
@@ -0,0 +1,718 @@
+                },
+            )
+            .returns(ReturnsNewTokenIdentifier)
+            .prepare_async()
+            .run()
+            .await;
+
+        println!("TOKEN ID: {:?}", res);
+    }
+
+    async fn issue_non_fungible_collection(
+        &mut self,
+        issue_cost: RustBigUint,
+        token_display_name: &str,
+        token_ticker: &str,
+    ) {
+        println!("Issuing NFT Collection...");
+        let nft_collection_id = self
+            .interactor
+            .tx()
+            .from(&self.wallet_address)
+            .to(ESDTSystemSCAddress.to_managed_address())
+            .gas(100_000_000u64)
+            .typed(ESDTSystemSCProxy)
+            .issue_non_fungible(
+                issue_cost.into(),
+                &token_display_name.into(),
+                &token_ticker.into(),
+                NonFungibleTokenProperties {
+                    can_freeze: true,
+                    can_wipe: true,
+                    can_pause: true,
+                    can_transfer_create_role: true,
+                    can_change_owner: true,
+                    can_upgrade: true,
+                    can_add_special_roles: true,
+                },
+            )
+            .returns(ReturnsNewTokenIdentifier)
+            .prepare_async()
+            .run()
+            .await;
+
+        println!("NFT Collection ID: {:?}", nft_collection_id);
+    }
+
+    async fn issue_semi_fungible_collection(
+        &mut self,
+        issue_cost: RustBigUint,
+        token_display_name: &str,
+        token_ticker: &str,
+    ) {
+        println!("Issuing SFT Collection...");
+        let sft_collection_id = self
+            .interactor
+            .tx()
+            .from(&self.wallet_address)
+            .to(ESDTSystemSCAddress.to_managed_address())
+            .gas(100_000_000u64)
+            .typed(ESDTSystemSCProxy)
+            .issue_semi_fungible(
+                issue_cost.into(),
+                &token_display_name.into(),
+                &token_ticker.into(),
+                SemiFungibleTokenProperties {
+                    can_freeze: true,
+                    can_wipe: true,
+                    can_pause: true,
+                    can_transfer_create_role: true,
+                    can_change_owner: true,
+                    can_upgrade: true,
+                    can_add_special_roles: true,
+                },
+            )
+            .returns(ReturnsNewTokenIdentifier)
+            .prepare_async()
+            .run()
+            .await;
+
+        println!("SFT Collection ID: {:?}", sft_collection_id);
+    }
+
+    async fn issue_token(
+        &mut self,
+        issue_cost: RustBigUint,
+        token_display_name: &str,
+        token_ticker: &str,
+        num_decimals: usize,
+        token_type: EsdtTokenType,
+    ) {
+        println!(
+            "Registering token... Don't forget to mint afterwards in order to receive the tokens!"
```

Comment:
Let's keep it simple, we are interested in showing what operations are done, the order of the operations and details are present in the docs